### PR TITLE
Some janitoring

### PR DESCRIPTION
**Pull request by akva2**
Quells some warnings with -Wshadow and -Wextra.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: ewoms/common/parametersystem.hh [hunk 1]
```diff
@@ -943,25 +943,25 @@ public:
...
+                      bool errorIfNotRegistered = true)
     {
 
 #ifndef NDEBUG
         // make sure that the parameter is used consistently. since
         // this is potentially quite expensive, it is only done if
         // debugging code is not explicitly turned off.
         check_(Dune::className<ParamType>(), propTagName, paramName);
+#else
+        (void) propTagName;
```
Comment (andlaus) [inline ewoms/common/parametersystem.hh]: please mark these parameters with OPM_DEBUG_UNUSED in the signature above instead.

File: ewoms/models/blackoil/blackoilenergymodules.hh [hunk 2]
```diff
@@ -233,7 +233,7 @@ public:
      * \brief Assign the energy specific primary variables to a PrimaryVariables object
      */
     static void assignPrimaryVars(PrimaryVariables& priVars,
-                                  Scalar temperature)
+                                  Scalar)
```
Comment (andlaus) [inline ewoms/models/blackoil/blackoilenergymodules.hh]: use OPM_UNUSED. (This allows to properly document parameters using doxygen and also enables to find out what their intention is.)

File: ewoms/common/parametersystem.hh (hunk 1 above)
Comment (andlaus) [reply ewoms/common/parametersystem.hh]: I mean OPM_OPTIM_UNSED...

**Comment by akva2** [comment]
updated.

**Comment by akva2** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this opm-common=858 please

**Comment by andlaus** [comment]
jenkins is happy, as expected. thanks for the patch!

**Pull request merged**